include/conversation.php

```diff
@@ -467,24 +469,27 @@ function item_fieldlists() {
 
 /**
  * @brief SQL join for contacts that are needed for displaying items
+ *
+ * @param int $uid user id
  */
-function item_joins() {
+function item_joins($uid = 0) {
 	return sprintf("STRAIGHT_JOIN `contact` ON `contact`.`id` = `item`.`contact-id`
 		AND NOT `contact`.`blocked`
 		AND ((NOT `contact`.`readonly` AND NOT `contact`.`pending` AND (`contact`.`rel` IN (%s, %s)))
 		OR `contact`.`self` OR (`item`.`id` != `item`.`parent`) OR `contact`.`uid` = 0)
 		INNER JOIN `contact` AS `author` ON `author`.`id`=`item`.`author-id` AND NOT `author`.`blocked`
 		INNER JOIN `contact` AS `owner` ON `owner`.`id`=`item`.`owner-id` AND NOT `owner`.`blocked`
+		LEFT JOIN `user-item` ON `user-item`.`iid` = `item`.`id` AND `user-item`.`uid` = %d
```

Comment:
I assume you used a `LEFT JOIN` because the `user-item` table isn't populated with the appropriate relations but that it will become authoritative later on, am I right? If yes, I'd rather we populate the table on DB update and use INNER JOIN instead.

Reply:
We had to create some update job to add an entry to every public item. This is a job that I really don't want to do so close to the new release.

Reply:
After more reading, it looks like `user-item` is currently only holding an optional `hidden` flag so this update job isn't warranted at the moment.

I thought it was to extract `uid` from the `item` table.

Reply:
In the future it will. But not now. This would be a much more complicated stuff.

Reply:
I know, that's why I first wrote what I did.